framework/i18n/Formatter.php

```diff
@@ -787,6 +787,51 @@ public function asRelativeTime($value, $referenceTime = null)
         }
     }
 
+    /**
+     * Formats the value as a time interval.
+     * @param \DateInterval|string|integer $value the value to be formatted, in ISO8601 format or number of seconds.
+     * @return string the formatted result.
+     */
+    public function asDuration($value)
+    {
+        if ($value === null) {
+            return $this->nullDisplay;
+        }
+        if ($value instanceof \DateInterval) {
+            $negative = $value->invert;
+            $interval = $value;
+        } elseif (is_numeric($value)) {
+            $negative = $value < 0;
+            $interval = new \DateInterval('PT' . abs($value) . 'S');
```

Comment:
Just found, that this will not work properly. You will have only count of seconds instead of splitting to days, hours, minutes etc.

```php
$interval = new \DateInterval('PT300S');
echo $interval->s; // 300
echo $interval->m; // 0
```